evalml/automl/automl_search.py

```diff
@@ -281,6 +285,11 @@ def __init__(
             objective = get_default_primary_search_objective(self.problem_type.value)
         objective = get_objective(objective, return_instance=False)
         self.objective = self._validate_objective(objective)
+        self.thresholding_objective = get_objective(
+            thresholding_objective, return_instance=True
+        )
+        if self.thresholding_objective.score_needs_proba:
+            raise ValueError("Thresholding objective must be thresholdable!")
```

Comment:
Nitpick, I love this error message but should we perhaps specify that the thresholding objective cannot be one that needs proba so it might be more clear to the user? They might pass an additional objective not fully knowing what it entails.